packages/framework/src/resources/workflow.test.ts

```diff
@@ -171,6 +171,75 @@ describe('workflow function', () => {
         });
     });
 
+    it('should not compile when the payload is not specified and the payloadSchema declares required properties', async () => {
+      const testWorkflow = workflow(
+        'test-workflow',
+        async ({ step, payload }) => {
+          await step.custom('custom', async () => ({
+            foo: 'bar',
+          }));
+        },
+        {
+          payloadSchema: {
+            type: 'object',
+            properties: {
+              foo: { type: 'string' },
+            },
+            required: ['foo'],
+            additionalProperties: false,
+          } as const,
+        }
+      );
+
+      // Capture in a test function to avoid throwing execution errors
+      const testFn = () =>
+        testWorkflow.trigger({
+          // @ts-expect-error - payload is missing from the trigger
+          payload: undefined,
+          to: 'test@test.com',
+        });
+    });
+
+    it('should compile when the payload is not specified and the payloadSchema does not declare required properties', async () => {
+      const testWorkflow = workflow(
+        'test-workflow',
+        async ({ step, payload }) => {
+          await step.custom('custom', async () => ({
+            foo: 'bar',
+          }));
+        },
+        {
+          payloadSchema: {
+            type: 'object',
+            properties: {
+              foo: { type: 'string' },
+            },
+            additionalProperties: false,
+          } as const,
+        }
+      );
+
+      // Capture in a test function to avoid throwing execution errors
+      const testFn = () =>
+        testWorkflow.trigger({
+          to: 'test@test.com',
+        });
+    });
+
+    it('should compile when the payload is not specified and the payloadSchema is not specified', async () => {
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
❓ **question:** It's debatable whether payload should be omitted entirely if payloadSchema is not present. It's a move towards greater strictness + type safety in Framework, but I wonder if we should be this opinionated and force specification of payloadSchema for those that would just like to trigger a workflow quickly.‏